WordPress/Classes/ViewRelated/Site Creation/Web Address/AddressTableViewCell+ViewModel.swift

```diff
@@ -0,0 +1,125 @@
+            "domain.suggestions.row.free",
+            value: "Free",
+            comment: "The text to display for free domains in 'Site Creation > Choose a domain' screen"
+        )
+        static let yearly = NSLocalizedString(
+            "domain.suggestions.row.yearly",
+            value: "per year",
+            comment: "The text to display for paid domains in 'Site Creation > Choose a domain' screen"
+        )
+        static let firstYear = NSLocalizedString(
+            "domain.suggestions.row.first-year",
+            value: "for the first year",
+            comment: "The text to display for paid domains on sale in 'Site Creation > Choose a domain' screen"
+        )
+        static let recommended = NSLocalizedString(
+            "domain.suggestions.row.recommended",
+            value: "Recommended",
+            comment: "The 'Recommended' label under the domain name in 'Choose a domain' screen"
+        )
+        static let bestAlternative = NSLocalizedString(
+            "domain.suggestions.row.best-alternative",
+            value: "Best Alternative",
+            comment: "The 'Best Alternative' label under the domain name in 'Choose a domain' screen"
+        )
+        static let sale = NSLocalizedString(
+            "domain.suggestions.row.sale",
+            value: "Sale",
+            comment: "The 'Sale' label under the domain name in 'Choose a domain' screen"
+        )
+    }
+}
+
+extension AddressTableViewCell.ViewModel {
+
+    init(model: DomainSuggestion, tags: [Tag] = []) {
+        // Declare variables
+        var tags = tags
+        let cost: Cost
+
+        // Format cost and sale cost
+        if model.isFree {
+            cost = .free
+        } else if let formatter = Self.currencyFormatter(code: model.currencyCode),
+                  let costValue = model.cost,
+                  let formattedCost = formatter.string(from: .init(value: costValue)) {
+            if let saleCost = model.saleCost, let formattedSaleCost = formatter.string(from: .init(value: saleCost)) {
+                cost = .onSale(cost: formattedCost, sale: formattedSaleCost)
+            } else {
+                cost = .regular(cost: formattedCost)
+            }
+        } else {
+            cost = .regular(cost: model.costString)
+        }
+
+        // Configure tags
+        if case .onSale = cost {
+            tags.append(.sale)
+        }
+
+        // Initialize instance
+        self.init(
+            domain: model.domainName,
+            tags: tags,
+            cost: cost
+        )
+    }
+
+    /// Returns a list of tags depending on the row's position in the list.
+    /// - Parameter position: The position of the domin suggestion in the list.
+    /// - Returns: A list of tags.
+    static func tagsFromPosition(_ position: Int) -> [Tag] {
+        switch position {
+        case 0: return [.recommended]
+        case 1: return [.bestAlternative]
+        default: return []
```

Comment:
The `sale` option is skipped. Is it intentional? Could defining the enum as an Int (`enum Tag: Int`) help?

Reply:
The requirement is to always display the "Recommended" label in the 1st domain suggestion, and "Best Alternative" in the 2nd domain suggestion. "Sale" label doesn't depend on the domain suggestion position, it is visible when a domain is on sale.

Adding an `Int` RawValue is valid when there is a clear order between enum cases. For example:

```swift
enum Week: Int {
   case monday = 0
   case tuesday
   case wednesday
   // etc...
}
```

In the previous example, it is clear that Tuesday comes after Monday and Wednesday comes after Tuesday etc...

But that's not the case with `Recommended`, `Best Alternative` and `Sale`.